### ISSUE TITLE
[Select] Listbox in `sl-select` does not have accessible name and is not referenced with `aria-controls`

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3435]
### 😯 Current Behavior

`sl-select-button` is not referencing listbox with `aria-controls`
`sl-listbox` does not have accessible name provided with `aria-labelledby` or `aria-label`

<img width="1033" height="830" alt="Image" src="https://github.com/user-attachments/assets/7aaef1c3-336b-4371-97bd-1dbfd4c05a39" />

### 🤔 Expected Behavior/possible solution?

`sl-select-button` should have `aria-controls='LISTBOX_ID'` to reference to listbox
`sl-listbox` should have `aria-labelledby='SELECT_LABEL_ID'` to have accessible name that is label of select element. 

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/form-select--basic
2. Expand listbox
3. Go to Accessibility tab in Storybook add-ons and refresh tests
4. Check if 'ARIA input field name' and 'ARIA required attributes' are present in test report

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/form-select--basic

### ⚖️ Impact

Not a huge problem, we have a workaround

### Framework

Other

### What browsers are you seeing the problem on?

All

### Component version

2.1.4

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS
